homeassistant/components/matter/__init__.py

```diff
@@ -195,6 +195,16 @@ async def async_remove_config_entry_device(
     if node is None:
         return True
 
+    if node.is_bridge_device:
+        config_entry = hass.config_entries.async_entries(DOMAIN)[0]
+        device_registry = dr.async_get(hass)
+        devices = dr.async_entries_for_config_entry(
+            device_registry, config_entry.entry_id
+        )
+        for device in devices:
+            if device.via_device_id == device_entry.id:
+                device_registry.async_remove_device(device.id)
```

Comment:
It's safer to remove the config entry id from the device via the `async_update_device` method.

https://github.com/home-assistant/core/blob/1fa82fa886c68aab504f9b055bedec1ca0b39a82/homeassistant/helpers/device_registry.py#L442

If it's the last config entry that is removed the device will be removed too.

Reply:
This wasn't addressed.

Reply:
sorry forgot to push the changes, should be fixed now